repair/row_level.cc

```diff
@@ -2770,6 +2793,36 @@ class row_level_repair {
         master.stats().round_nr_slow_path++;
     }
 
+private:
+    // Update system.repair_history table
+    future<> update_system_repair_table() {
+        // Update repair_history table only if it is a reguar repair.
+        if (_ri.reason != streaming::stream_reason::repair) {
+            co_return;
+        }
+        // Update repair_history table only if all replicas have been repaired
+        size_t repaired_replicas = _all_live_peer_nodes.size() + 1;
+        if (_ri.total_rf != repaired_replicas){
+            rlogger.debug("repair[{}]: Skipped to update system.repair_history total_rf={}, repaired_replicas={}, local={}, peers={}",
+                    _ri.id.uuid, _ri.total_rf, repaired_replicas, utils::fb_utilities::get_broadcast_address(), _all_live_peer_nodes);
+            co_return;
+        }
+        repair_service& rs = _ri.rs;
+        std::optional<gc_clock::time_point> repair_time_opt = co_await rs.update_history(_ri.id.uuid, _table_id, _range, _start_time);
+        if (!repair_time_opt) {
+            co_return;
+        }
+        auto repair_time = repair_time_opt.value();
+        repair_update_system_table_request req{_ri.id.uuid, _table_id, _ri.keyspace, _cf_name, _range, repair_time};
+        auto all_nodes = _all_live_peer_nodes;
+        all_nodes.push_back(utils::fb_utilities::get_broadcast_address());
+        co_await parallel_for_each(all_nodes, [this, req] (gms::inet_address node) -> future<> {
+            repair_update_system_table_response resp = co_await _ri.messaging.local().send_repair_update_system_table(netw::messaging_service::msg_addr(node), req);
+            rlogger.debug("repair[{}]: Got repair_update_system_table_response from node {}", _ri.id.uuid, node);
+        });
```

Comment:
This will throw if one of the nodes is dead, I think we need to log & swallow since it doesn't affect repair correctness.

Reply:
Yes, this makes sense.

Reply:
Done.